railties/lib/rails/generators/rails/resource_route/resource_route_generator.rb

```diff
@@ -21,7 +25,9 @@ def add_resource_route
         end
 
         # inserts the primary resource
-        write("resources :#{file_name.pluralize}", route_length + 1)
+        resource_string = "resources :#{file_name.pluralize}"
+        resource_string << ", except: [:new, :edit]" if options.api?
```

Comment:
This will always generate `resources :post, except: [:new, :edit]`, even when  `resources` generate only 5 routes in a API-only apps, excluding `new` and `edit` out-of-the-box.

It concerns me because generated code should be as simple as possible, and it's odd to generate a line in the route file excluding the actions that are excluded internally.

If we're still interested on supporting the `--api` in generators, and it could be different that the global api-only flag, I would try to generate the line with the `except` part **only** when `Rails.application.config.api_only` is false

/cc @rafaelfranca